examples/plot_mdm_spd_helper.py

```diff
@@ -0,0 +1,145 @@
+        self.n_features=n_features
+        self.n_classes=n_classes
+        self.data_helper=DataHelper()
+
+    def generate_sample_dataset(self):
+        X,Y=self.easy_gauss()
+        X,Y=self.data_helper.shuffle(X,Y)
+        return X,Y
+
+    def mat2vec(self,mat):
+        n=mat.shape[-1]
+        return mat[numpy.tril_indices(n)]
+
+    # returns one random rotation of dimension n
+    def random_rotation(self):
+        if(self.n_features==2):
+            theta=numpy.random.uniform(0,2*numpy.pi)
+            c,s=numpy.cos(theta),numpy.sin(theta)
+            s=numpy.array([[c,-s],[s,c]])
+            return s
+        z=(numpy.random.randn(n,n)+1j*numpy.random.randn(n,n))/numpy.sqrt(2.0)
+        q,r=numpy.linalg.qr(z)
+        d=numpy.diag(r)
+        ph=numpy.diag(d/numpy.abs(d))
+        # q=numpy.dot(q,numpy.dot(ph,q))
+        s=numpy.dot(q,ph)
+        return s
+
+    # returns N random rotation matrices close of var_rot around R
+    def random_rotations(self,R,var_rot):
+        rots=numpy.zeros((self.n_samples,self.n_features,self.n_features))
+        for i in range(self.n_samples):
+            rots[i]=numpy.dot(R,self.divide_angle_of_cov2(self.random_rotation(),var_rot))
+        return rots
+
+    def angle_of_rot2(self,r):
+        return numpy.arctan(r[0][1]/r[0][0])
+
+    def divide_angle_of_cov2(self,r,alpha):
+        angle=self.angle_of_rot2(r)*alpha
+        c,s=numpy.cos(angle),numpy.sin(angle)
+        return numpy.array([[c,-s],[s,c]])
+
+    def easy_gauss(self):
+
+        #hyperparams
+        M=numpy.random.uniform(-5,5,(self.n_classes,self.n_features)) #get self.n_classes mean mean vectors
+        S=numpy.random.uniform(0.1,5.,(self.n_classes,self.n_features)) #get self.n_classes mean diagonal covariances
+        if(self.n_features==2):
+            R=self.random_rotations(numpy.eye(self.n_features),1) #get self.n_classes mean rotations
+        var_mean=numpy.eye(self.n_features)*0.05 #class variance in mean
+        var_cov=numpy.eye(self.n_features)*0.05 #in covariance
+        if(self.n_features==2):
+            var_rot=0.01 #in rotation
+
+        #data
+        mu=numpy.zeros((self.n_classes*self.n_samples,self.n_features))
+        cov=numpy.zeros((self.n_classes*self.n_samples,self.n_features,self.n_features))
+        Y=numpy.zeros((self.n_classes*self.n_samples,self.n_classes))
+        for i in range(self.n_classes):
+            means=numpy.random.multivariate_normal(M[i],var_mean,self.n_samples)
+            covs=numpy.random.multivariate_normal(S[i],var_cov,self.n_samples)
+            rots=self.random_rotations(R[i],var_rot)
+            mu[i*self.n_samples:(i+1)*self.n_samples]=means
+            for j in range(self.n_samples):
+                c=numpy.diag(numpy.abs(covs[j]))
+                c=numpy.dot(rots[j],numpy.dot(c,rots[j].T))
+                cov[i*self.n_samples+j]=c
+            Y[i*self.n_samples:(i+1)*self.n_samples,i]=1
+        return cov,Y
+
+class PlotHelper():
```

Comment:
This could be exposed in the `geomstats/visualization` module? It would make this code easier to find for users.

Reply:
Following the logic in the file `visualization.py`, you could put that code in a class called `Ellipsis`, that would have methods `draw` and `plot`.

Reply:
I ported the relevant parts to `visualization.py` to an `Ellipsis` class, tried to follow the logic of the the other classes.